std/bitmanip.d

```diff
@@ -247,12 +248,39 @@ private template createTaggedReference(T, ulong a, string name, Ts...)
 }
 
 /**
-Allows creating bit fields inside $(D_PARAM struct)s and $(D_PARAM
-class)es.
+Allows creating `bitfields` inside `structs`, `classes` and `unions`.
 
-The type of a bit field can be any integral type or enumerated
-type. The most efficient type to store in bitfields is $(D_PARAM
-bool), followed by unsigned types, followed by signed types.
+A `bitfield` consists of one or more entries with a fixed number of
+bits reserved for each. The types of the enties can be `bool`s,
+integral types or enumerated types, arbitrarily mixed.
+
+Integral and Enumerated types are accompanied by two `enums`: `min`
+and `max`, giving the minimum and the maximum number, that can be
+represented by this entry. The variable names are preceded by the
+name of the variable followed by an underscore, so variable `x` comes
+along with `x_min` and `x_max`.
+
+The most efficient type to store in `bitfields` is `bool`, followed by
+unsigned types, followed by signed types.
+
+Limitation: The number of bits in a `bitfield` is limited to 8, 16,
+32 or 64. If padding is needed, just allocate an entry with an empty
+name.
+
+Implementation_details: `Bitfields` are internally stored in an
+`ubyte`, `ushort`, `uint` or `ulong` depending on the number of bits
+used. The bits are filled in in order given by the parameters,
```

Comment:
I think that the information given from this sentence to the end of the paragraph is too much. Specifying the internal storage should be enough.

Reply:
I disagree here. For example, have a look at FloatRep further down this file. If there is no guarantee on the order of the entries, a user cannot reliably do such things.

The name of the private variable, well yes, probably unnecessary. I thought it might prevent a name clash, to know about it. But I must admit, that this is very unlikely. And I checked the error message, it directly points to the mixin as the cause, so people might infere the reason without having to peek into the source code.

Reply:
Yeah, that makes sense. The please update the double in.
```suggestion
used. The bits are filled in the order given by the parameters,
```